src/hooks/useContentManager.js

```diff
@@ -0,0 +1,211 @@
+        });
+    },
+    [schema]
+  );
+
+  const handleChange = useCallback(
+    (event) => {
+      const { name, value } = event.target;
+      setPageConfig((prevPageConfig) => {
+        return {
+          ...prevPageConfig,
+          [name]: value
+        };
+      });
+      validateField(name, value);
+    },
+    [validateField, setPageConfig]
+  );
+  const savePageConfig = useCallback(() => {
+    validateObject(pageConfig);
+    axiosPrivate
+      .post(`/api/protected/${currentConfig.pageConfig}`, pageConfig)
+      .then(({ data }) => {
+        console.log("saved succefylly ", data);
+      })
+      .catch((err) => {
+        console.error("error :>> ", err);
+      });
+  }, [pageConfig, validateObject]);
+  const handleTabChange = useCallback(
+    (event, newValue) => {
+      setActiveTab(newValue);
+    },
+    [setActiveTab]
+  );
+
+  const handleAddModalOpen = useCallback((title) => {
+    setPageDialogTitle(title);
+    setModalOpenAdd(() => true);
+  }, []);
+  const handleAddModalClose = useCallback(() => {
+    setModalOpenAdd(false);
+    setPageConfig(pageConfigInitial);
+  }, [setModalOpenAdd, setPageConfig]);
+
+  const pageConfigFormProps = useMemo(() => {
+    return { pageConfig: { ...pageConfig }, handleChange, errors };
+  }, [errors, handleChange, pageConfig]);
+
+  const updatePageConfig = useCallback(() => {
+    validateObject(pageConfig);
+    axiosPrivate
+      .put(`/api/protected/${currentConfig.pageConfig}`, pageConfig)
+      .then(({ data }) => {
+        console.log("saved succefylly ", data);
+      })
+      .catch((err) => {
+        console.error("error :>> ", err);
+      });
+    setPageConfig(pageConfigInitial);
+  }, [validateObject, pageConfig]);
+  const populatePageConfigForm = useCallback(
+    (row) => {
+      const { id, name, page_type, parent, language, header_text, img_link, header_image, order_number, description } =
+        row.original;
+
+      const pageConfigTemp = {
+        id: id,
+        pageType: page_type,
+        parent: parent,
+        headerText: header_text,
+        language: language,
+        imageLink: img_link,
+        headerImage: header_image,
+        orderNumber: order_number,
+        description,
+        name
+      };
+
+      setPageConfig((prevPageConfig) => {
+        return {
+          ...prevPageConfig,
+          ...pageConfigTemp
+        };
+      });
+      handleAddModalOpen("Update Page Config");
+    },
+    [handleAddModalOpen]
+  );
+
+  const deletePageConfig = useCallback((row) => {
+    const { id } = row.original;
+    axiosPrivate
+      .delete(`/api/protected/${currentConfig.pageConfig}/${id}`)
+      .then(({ data }) => {
+        console.log("data deleted  ", data.id);
+      })
+      .catch((err) => {
+        console.error("error :>> ", err);
+      });
+  }, []);
+
+  const dialogFormProps = useMemo(() => {
+    return {
+      0: {
+        dialogProps: { ...pageConfigFormProps, pageConfig: { ...pageConfig } },
```

Comment:
if these are nested object then this is just creating a shallow copy so if have nested object then please reconsider this destructuring